numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -3105,22 +3105,24 @@ OBJECT_argmax(PyObject **ip, npy_intp n, npy_intp *max_ind,
     PyObject *mp = ip[0];
 
     *max_ind = 0;
-    i = 1;
-    while (i < n && mp == NULL) {
+    for (i = 1; i < n && mp == NULL; ++i) {
```

Comment:
I think things will be a bit simpler if the loop looks like
```
for (i = 1; i < n && ip[i] == NULL; ++i);
```
`i` will be the index of the first non-NULL on exit. At that point you can set the values for all null and return, Or set `mp` and `max_ind` to continue. 

The original looks like a mutant cross between two species of loop ;)